tower-http/src/services/fs/serve_dir.rs

```diff
@@ -245,9 +251,25 @@ impl<ReqBody> Service<Request<ReqBody>> for ServeDir {
                     HeaderValue::from_str(mime::APPLICATION_OCTET_STREAM.as_ref()).unwrap()
                 });
 
-            let (file, maybe_encoding) =
+            let (mut file, maybe_encoding) =
                 open_file_with_fallback(full_path, negotiated_encodings).await?;
-            Ok(Output::File(file, mime, buf_chunk_size, maybe_encoding))
+            let file_size = file.metadata().await?.len();
+            let maybe_range = range_header.map(|range| parse_range(&range, file_size));
+            if let Some(ParsedRangeHeader::Range(ranges)) = maybe_range.as_ref() {
+                if ranges.len() == 1 {
```

Comment:
Is it correct that this means if the client asks for multiple ranges we'll only return the first one? Not an expert on the spec for `Range` but is that allowed? Would returning some sort of "range not satisfiable" status be more correct?

I remember that supporting multiples ranges is quite annoying so I'm fine with not supporting it. Its something we should document regardless.

Reply:
Edit: Ah I see you're handling multipart ranges further down. Adding a comment about that here would be fine I think.